src/Database/Database.php

```diff
@@ -2919,15 +2919,73 @@ public function updateDocument(string $collection, string $id, Document $documen
         if ($collection->getId() !== self::METADATA) {
             $documentSecurity = $collection->getAttribute('documentSecurity', false);
 
-            $relationshipKeys = [];
             foreach ($relationships as $relationship) {
                 $relationshipKey = $relationship->getAttribute('key');
-                $relationshipKeys[$relationshipKey] = $relationshipKey;
+                $relationships[$relationshipKey] = $relationship;
             }
             // Compare if the document has any changes
             foreach ($document as $key=>$value) {
                 // Skip the nested documents as they will be checked later in recursions.
-                if (array_key_exists($key, $relationshipKeys)) {
+                if (array_key_exists($key, $relationships)) {
+                    $relationType = (string) $relationships[$key]['options']['relationType'];
+                    $side = (string) $relationships[$key]['options']['side'];
+                    switch($relationType) {
+                        case Database::RELATION_ONE_TO_ONE: {
+                            $oldValue = $old->getAttribute($key) instanceof Document ? $old->getAttribute($key)->getId() : $old->getAttribute($key);
```

Comment:
Let's line break to make ternarys easier to read:

```suggestion
                            $oldValue = $old->getAttribute($key) instanceof Document 
                                ? $old->getAttribute($key)->getId()
                                : $old->getAttribute($key);
```